builtin/providers/google/resource_container_cluster.go

```diff
@@ -342,19 +342,21 @@ func resourceContainerClusterCreate(d *schema.ResourceData, meta interface{}) er
 	zoneName := d.Get("zone").(string)
 	clusterName := d.Get("name").(string)
 
-	masterAuths := d.Get("master_auth").([]interface{})
-	if len(masterAuths) > 1 {
-		return fmt.Errorf("Cannot specify more than one master_auth.")
+	cluster := &container.Cluster{
+		Name:             clusterName,
+		InitialNodeCount: int64(d.Get("initial_node_count").(int)),
 	}
-	masterAuth := masterAuths[0].(map[string]interface{})
 
-	cluster := &container.Cluster{
-		MasterAuth: &container.MasterAuth{
+	if v, ok := d.GetOk("master_auth"); ok {
+		masterAuths := v.([]interface{})
+		if len(masterAuths) > 1 {
```

Comment:
While you're here, would you mind removing this check and instead putting MaxItems: 1 in the schema defined above? That'll move the check from the apply phase to the plan phase.